dbms/src/Storages/Transaction/KVStore.cpp

```diff
@@ -333,12 +330,27 @@ void KVStore::setRegionCompactLogConfig(UInt64 sec, UInt64 rows, UInt64 bytes)
         bytes);
 }
 
-void KVStore::persistRegion(const Region & region, const RegionTaskLock & region_task_lock, const char * caller)
+void KVStore::persistRegion(const Region & region, std::optional<const RegionTaskLock *> region_task_lock, const char * caller)
 {
-    LOG_INFO(log, "Start to persist {}, cache size: {} bytes for `{}`", region.toString(true), region.dataSize(), caller);
     RUNTIME_CHECK_MSG(region_persister, "try access to region_persister without initialization, stack={}", StackTrace().toString());
-    region_persister->persist(region, region_task_lock);
-    LOG_DEBUG(log, "Persist {} done", region.toString(false));
+    if (region_task_lock.has_value())
+    {
+        LOG_INFO(log, "Start to persist {}, cache size: {} bytes for `{}`", region.toString(true), region.dataSize(), caller);
+        region_persister->persist(region, *region_task_lock.value());
+    }
+    else
+    {
+        LOG_INFO(log, "Try to persist {}", region.toString(false));
+        region_persister->persist(region);
+    }
+    if (region_task_lock.has_value())
+    {
+        LOG_DEBUG(log, "Persist {} done", region.toString(false));
+    }
+    else
+    {
+        LOG_INFO(log, "After persisted {}, cache {} bytes", region.toString(false), region.dataSize());
+    }
```

Comment:
the statements inside this if branch can be merged into the previous if branch?

Reply:
fixed